lib/boom_notifier/error_storage.ex

```diff
@@ -81,53 +75,59 @@ defmodule BoomNotifier.ErrorStorage do
   """
   @spec send_notification?(ErrorInfo.t()) :: boolean()
   def send_notification?(error_info) do
-    %{key: error_hash_key} = error_info
-
-    error_storage_item =
-      Agent.get(:boom_notifier, fn state -> state end)
-      |> Map.get(error_hash_key)
-
-    do_send_notification?(error_storage_item)
+    error_info
+    |> get_error_stats()
+    |> do_send_notification?()
   end
 
   @doc """
   Reset the accumulated_occurrences for the given error info to zero. It also
   increments the max storage capacity based on the notification strategy.
+
+  Returns error storage entry before reset
   """
-  @spec reset_accumulated_errors(error_strategy, ErrorInfo.t()) :: :ok
-  def reset_accumulated_errors(:exponential, error_info) do
-    %{key: error_hash_key} = error_info
+  @spec reset(ErrorInfo.t()) :: __MODULE__.t()
```

Comment:
Loved the rename of this function 😍

Reply:
What about something like `ErrorStorage.reset_stats/2` or `ErrorStorage.reset_and_get_stats/2`

I'm thinking about the `notifications_sender` reading:

```elixir
occurrences = ErrorStorage.reset_stats(error_info, notification_trigger)
```

Reply:
makes sense to be consistent with other ErrorStorage's functions